classes/blocks/class-blocklist.php

```diff
@@ -73,16 +73,16 @@ public static function get_block_list( ?int $post_id = null, &$reusable = [] ):
 	 *
 	 * @return string[] List of unique block names.
 	 */
-	public static function parse_block_list( string $content, &$seen = [], $post_id = 0, bool $include_articles ): array {
+	public static function parse_block_list( string $content, &$seen = [], $post_id = 0, bool $include_articles = false ): array {
 		// Add Articles block in posts where it's included from the settings.
-		$initial_blocks = $include_articles ? [ 'planet4-blocks/articles' ] : [];
+		$initial_blocks = $include_articles ? [ 'blockName' => 'planet4-blocks/articles' ] : [];
 
 		if ( ! has_blocks( $content ) ) {
 			return $initial_blocks;
 		}
 
 		$content_blocks = ( new WP_Block_Parser() )->parse( $content );
-		$blocks         = array_merge( $initial_blocks, $content_blocks );
+		$blocks         = array_merge( [ $initial_blocks ], $content_blocks );
```

Comment:
I don't understand why we need to add extra brackets here, since `$initial_blocks` is already an array? 🤔

Reply:
That's required because the end result needs to be an array of arrays. Each block is represented as an array of its own.

You can see it if you dump the contents of `$blocks`. And then after line 86, `$parsed` ends up not containing `planet4-blocks/articles` at all. So just merging `$initial_blocks` is resulting in "Related Articles" not being displayed in the frontend.